blues/models.py

```diff
@@ -0,0 +1,317 @@
+    b: 1x3 np.array
+        Defines position of particle to be transformed into
+        new basis set.
+    Returns
+    -------
+    changed_coord: 1x3 np.array
+        Coordinates of b in new basis.
+    '''
+    ainv = np.linalg.inv(a.T)
+    changed_coord = np.dot(ainv,b.T)*unit.nanometers
+    return changed_coord
+def undoBasis(a, b):
+    '''
+    Transforms positions in a transformed basis (b) to the regular
+    basis set.
+    Arguments
+    ---------
+    a: 3x3 np.array
+        Defines vectors that defined the new basis.
+    b: 1x3 np.array
+        Defines position of particle to be transformed into
+        regular basis set.
+    Returns
+    -------
+    changed_coord: 1x3 np.array
+        Coordinates of b in new basis.
+    '''
+    a = a.T
+    changed_coord = np.dot(a,b.T)*unit.nanometers
+    return changed_coord
+
+
+def normalize(vector):
+    '''Normalize a given vector
+    Arguemnts
+    ---------
+    vector: 1xn np.array
+        Vector to be normalized.
+    Returns
+    -------
+    unit_vec: 1xn np.array
+        Normalized vector.
+    '''
+    magnitude = np.sqrt(np.sum(vector*vector))
+    unit_vec = vector / magnitude
+    return unit_vec
+
+def localcoord(particle1, particle2, particle3):
+    '''Defines a new coordinate system using 3 particles
+    returning the new basis set vectors
+    '''
+    part1 = particle1 - particle1
+    part2 = particle2 - particle1
+    part3 = particle3 - particle1
+    vec1 = part2
+    vec2= part3
+    vec3 = np.cross(vec1,vec2)*unit.nanometers
+    return vec1, vec2, vec3
+
+def findNewCoord(particle1, particle2, particle3, center):
+    '''Finds the coordinates of a given center in a new coordinate
+        system defined by particles1-3
+    '''
+    #calculate new basis set
+    vec1, vec2, vec3 = localcoord(particle1, particle2, particle3)
+    basis_set = np.zeros((3,3))*unit.nanometers
+    basis_set[0] = vec1
+    basis_set[1] = vec2
+    basis_set[2] = vec3
+    #since the origin is centered at particle1 by convention
+    #subtract to account for this
+    recenter = center - particle1
+    #find coordinate in new coordinate system
+    new_coord = changeBasis(basis_set, recenter)
+    old_coord = undoBasis(basis_set, new_coord)
+    return new_coord
+
+def findOldCoord(particle1, particle2, particle3, center):
+    '''Finds the coordinates of a given center (defined by a different basis
+    given by particles1-3) back in euclidian coordinates
+        system defined by particles1-3
+    '''
+
+    vec1, vec2, vec3 = localcoord(particle1, particle2, particle3)
+    basis_set = np.zeros((3,3))*unit.nanometers
+    basis_set[0] = vec1
+    basis_set[1] = vec2
+    basis_set[2] = vec3
+    #since the origin is centered at particle1 by convention
+    #subtract to account for this
+    old_coord = undoBasis(basis_set, center)
+    adjusted_center = old_coord + particle1
+    return adjusted_center
+
+class Model_SmartDart(Model):
```

Comment:
I like the use of inheritance here, but this leaves me with some questions that maybe someone can provide some insight on. My understanding is that if we structure it like this, an object that is instanced with `lig = Model_SmartDart` would have the functionality related to getting properties for the random_rotation move (since it inherits from the base class `Model) but an object instanced with `lig = Model` would not have the functionality related to smart darting. 

I think this would make it tricky to combine moves down the line? Or I could be entirely wrong and need to think about this some more.